### DISCUSSION TITLE
Domain Specific Language POC

### DISCUSSION THREAD

**Discussion by Audavis** [square/kotlinpoet#1458] [Show and tell]
I've created a proof of concept for a Domain Specific Language (DSL) pattern for Kotlin Poet. I'm kind of curious what you folks think. If it's well received I'd be willing to build DSL for the remainder of the library.

The POC is just converting the short Hello World tutorial on the Kotlin Poet site into DSL.
Hello World Example: https://square.github.io/kotlinpoet/#example

Starting code using Chained Builders:
![image](https://user-images.githubusercontent.com/40345691/215249458-73e20369-d86a-4c8f-b4fa-1d9ec26fb65f.png)

Ending code using DSL:
![image](https://user-images.githubusercontent.com/40345691/215249782-03f1a095-2975-4d15-8b76-877c296b0c6a.png)

**Comment by ty1824** [comment]
As a big fan of Kotlin DSLs and KotlinPoet (I use KPoet to generate DSLs!!!) I would love something like this. KPoet is quite verbose and this reads much cleaner.
